src/Sarif/SarifExtensionMethods.cs

```diff
@@ -258,7 +259,19 @@ public static string GetFileName(this Uri uri)
 
         public static string GetFilePath(this Uri uri)
         {
-            return uri.IsAbsoluteUri ? uri.LocalPath : uri.OriginalString;
+            if (!uri.IsAbsoluteUri)
```

Comment:
C:\Local%20DFile\File.txt

Reply:
Path.GetFullPath(OriginalString)

Reply:
more test cases are added, including this style.

Reply:
Tried, but Path.GetFullPath(...) add a lot of unexpected prefix, which is not right.

Reply:
What about 
Path.Combine(
Path.GetDirectoryName(uri.OriginalString),
Path.GetFileName(uri.OriginalString));
?

Generally speaking: System.IO.Path is how .NET exposes ways of manipulating local file system paths which are expressed as strings.  (In a system-independent way.)  I believe such a string must be what was originally passed into the Uri constructor.  If that's right, then the most promising strategy is going to be trying to figure out how to leverage System.IO.Path to manipulate that original string into the form that we need.

Attempting to correctly reverse the encoding performed by the URI constructor ourselves may not be strictly impossible, but it entails a lot more risk of being bit again later by something we do not understand about the intersection between Uri encoding and/or file system paths on one of the platforms we target.